assets/app/view/game/route_selector.rb

```diff
@@ -258,6 +259,26 @@ def actions(render_halts)
           store(:routes, @routes)
         end
 
+        add_train = lambda do
+          @game.add_route_train(@routes)
+          store(:routes, @routes)
+        end
+
+        delete_train = lambda do
+          @game.delete_route_train(@selected_route)
+          store(:routes, @routes)
+        end
+
+        increase_train = lambda do
+          @game.increase_route_train(@selected_route)
```

Comment:
did you implement these methods? i don't see them, also how can the UI change the game state, you need an action representing this don't you

Reply:
Yes. They are implemented in game.rb

The UI only changes game state indirectly. It manipulates the trains and their distances attached to the routes submitted to the RunRoutes action. When the action is processed the game state train distances are updated via the game method update_route_trains.